langlib/langlib-test/src/test/resources/test-src/typedesclib_test.bal

```diff
@@ -40,27 +46,38 @@ public function testGeTypeIds() {
     assertSingleTypeId(k, ["Error", "MyOtherError"]);
 
     k = ErrorAndMyOtherErrorD.typeIds();
-    assertSingleTypeId(k, ["Error", "MyOtherError", "ErrorAndMyOtherErrorD", "Error"]);
+    assertSingleTypeId(k, ["Error", "MyOtherError", "ErrorAndMyOtherErrorD"]);
 
     k = ErrorAndMyOtherErrorD.typeIds(true);
-    assertSingleTypeId(k, ["Error", "MyOtherError", "ErrorAndMyOtherErrorD"]);
+    assertSingleTypeId(k, ["ErrorAndMyOtherErrorD"]);
+
+    k = TwoDistinctKeywords.typeIds();
+    assertSingleTypeId(k, [0, 1]);
+
+    k = ThreeDistinctKeyworlds.typeIds();
+    assertSingleTypeId(k, [2, 3, 4]);
+
+    k = ThreeDiscintKeyworldsWithGroup.typeIds();
```

Comment:
For a non distinct type this will give a NPE is this expected.
```
error: java.lang.NullPointerException
        at ballerina.lang.typedesc.1_0_0:typeIds(typedesc_lib.bal:46)
           rukshanp.TypeIds.0_1_0:main(main.bal:29)
```

Reply:
I can't reproduce the NPE, can you provide the sample code for this.
Yes, when the type is not definite, this should return nil, will fix this.

Reply:
Please ignore this. I tested again. This is only reproducible in master. Probably I was testing an old build or a failed build.